usage/src/org/apache/cloudstack/quota/QuotaAlertManagerImpl.java

```diff
@@ -0,0 +1,444 @@
+                }
+            } else {
+                s_logger.info("For " + quotaAccount.getId() + " it is already more than " + getDifferenceDays(lastStatementDate, new Date()) + " days, will send statement in next cycle");
+            }
+        }
+
+        for (DeferredQuotaEmail emailToBeSent : deferredQuotaEmailList) {
+            s_logger.debug("Attempting to send quota STATEMENT email to users of account: " + emailToBeSent.getAccount().getAccountName());
+            sendQuotaAlert(emailToBeSent);
+        }
+    }
+
+    @Override
+    public void checkAndSendQuotaAlertEmails() {
+        s_logger.info("Running checkAndSendQuotaAlertEmails");
+        List<DeferredQuotaEmail> deferredQuotaEmailList = new ArrayList<DeferredQuotaEmail>();
+        final BigDecimal zeroBalance = new BigDecimal(0);
+        for (final QuotaAccountVO quotaAccount : _quotaAcc.listAll()) {
+            s_logger.info("checkAndSendQuotaAlertEmails accId=" + quotaAccount.getId());
+            BigDecimal accountBalance = quotaAccount.getQuotaBalance();
+            Date balanceDate = quotaAccount.getQuotaBalanceDate();
+            Date alertDate = quotaAccount.getQuotaAlertDate();
+            int lockable = quotaAccount.getQuotaEnforce();
+            BigDecimal thresholdBalance = quotaAccount.getQuotaMinBalance();
+            if (accountBalance != null) {
+                AccountVO account = _accountDao.findById(quotaAccount.getId());
+                if (s_logger.isDebugEnabled()) {
+                    s_logger.debug("Check id " + account.getId() + " bal=" + accountBalance + " alertDate" + alertDate + " current date" + new Date());
+                }
+                if (accountBalance.compareTo(zeroBalance) <= 0) {
+                    if (_lockAccountEnforcement && (lockable == 1)) {
+                        if (account.getType() == Account.ACCOUNT_TYPE_NORMAL) {
+                            lockAccount(account.getId());
+                        }
+                    }
+                    if (alertDate == null || (balanceDate.after(alertDate) && getDifferenceDays(alertDate, new Date()) > 1)) {
+                        deferredQuotaEmailList.add(new DeferredQuotaEmail(account, quotaAccount, QuotaConfig.QuotaEmailTemplateTypes.QUOTA_EMPTY));
+                    }
+                } else if (accountBalance.compareTo(thresholdBalance) <= 0) {
+                    if (alertDate == null || (balanceDate.after(alertDate) && getDifferenceDays(alertDate, new Date()) > 1)) {
+                        deferredQuotaEmailList.add(new DeferredQuotaEmail(account, quotaAccount, QuotaConfig.QuotaEmailTemplateTypes.QUOTA_LOW));
+                    }
+                }
+            }
+        }
+
+        for (DeferredQuotaEmail emailToBeSent : deferredQuotaEmailList) {
+            s_logger.debug("Attempting to send quota alert email to users of account: " + emailToBeSent.getAccount().getAccountName());
```

Comment:
This log statement should wrapped in a ``if (s_logger.isDebugEnabled()) { } `` block.